client/lib/polyfill/index.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Internal dependencies
+ */
+import localStoragePolyfill from 'lib/local-storage';
+
+export default ( () => {
+	if ( 'undefined' === typeof window ) {
```

Comment:
I think you're right that we don't need or want the `export default`. The only other reason to keep it an IIFE is to allow the early return, since there can't be top-level returns in a module. Otherwise we could just put all the browser-specific polyfills inside the `if` block.

>Or just once on the first `require`?

This is the intended behavior.